pkg/scheduler/queue/queue.go

```diff
@@ -96,26 +98,29 @@ func (q *RequestQueue) EnqueueRequest(userID string, req Request, maxQueriers fl
 	}
 
 	shardSize := util.DynamicShardSize(maxQueriers, len(q.queues.queriers))
-	queue := q.queues.getOrAddQueue(userID, shardSize)
+	queue := q.queues.getOrAddQueue(userID, shardSize, req.IsHighPriority())
+	maxOutstandingRequests := q.queues.limits.MaxOutstandingPerTenant(userID)
+
 	if queue == nil {
 		// This can only happen if userID is "".
 		return errors.New("no queue found")
 	}
 
 	q.totalRequests.WithLabelValues(userID).Inc()
```

Comment:
Let's also add metric label to differentiate high/normal priority requests?
Same for all related metric labels